Sources/SwiftParser/StringLiterals.swift

```diff
@@ -593,6 +593,37 @@ extension Parser {
       )
     }
   }
+
+  mutating func parseSimpleString() -> RawSimpleStringLiteralExprSyntax {
+    let openDelimiter = self.consume(if: .rawStringDelimiter)
+    let (unexpectedBeforeOpenQuote, openQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpenQuoteOptions.self, default: .stringQuote)
+
+    /// Parse segments.
+    var segments: [RawStringSegmentSyntax] = []
+    var loopProgress = LoopProgressCondition()
+    while loopProgress.evaluate(self.currentToken) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
+
+      if let stringSegment = self.consume(if: .stringSegment, TokenSpec(.identifier, remapping: .stringSegment)) {
+        segments.append(RawStringSegmentSyntax(content: stringSegment, arena: self.arena))
+      }
```

Comment:
How difficult, do you think, would it be to also consume string interpolation segments here and make them unexpected? Or do you know how the current diagnostics look like if a simple string literals contains interpolation segments?